### PR TITLE
feat(anonymizer): pack-based detection, 16-hex tokens, validation & config

### PR DESCRIPTION
**Pull request by iamclaude697**
## Summary

- **Pack-based PII detection**: replaces regex-only scanning with a pluggable pack architecture (`internal/anonymizer/anonymizer.go`) for structured, composable detection rules
- **16-hex privacy tokens**: PII values are replaced with deterministic `[PII_XXXXXXXX]` tokens (8 hex chars) that survive round-trips through LLM providers and can be reversed on response
- **Validation layer**: input/output validation with Pydantic-style config validation in `internal/config/`
- **Streaming support**: `internal/anonymizer/streaming.go` handles chunked/streamed responses with token-boundary awareness
- **Reverse proxy core**: `internal/proxy/proxy.go` wires anonymization into the HTTP proxy pipeline
- **Configuration**: `proxy-config.json` provides runtime config; `internal/config/` loads and validates it
- **Metrics stub**: `internal/metrics/metrics.go` exposes anonymization counters
- **Documentation**: `docs/architecture.md` and `docs/configuration.md` describe the system design and config options

## Test plan

- [ ] `go test ./internal/config/...` — config loading & validation
- [ ] `go test ./internal/anonymizer/...` — pack detection, token generation, streaming edge cases
- [ ] `go test -race ./...` — race detector across all packages
- [ ] Manual: start proxy with `proxy-config.json`, send request with PII, verify tokens in forwarded request and de-anonymized response

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: anonymizer.go [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+
+	if a.verbose {
+		log.Printf("[DEANON] StreamingDeanonymize sessionID=%s tokens=%d", sessionID, len(tokenMap))
+	}
+	if len(tokenMap) == 0 {
+		return src
+	}
+
+	if a.m != nil {
+		a.m.TokensDeanonymized.Add(int64(len(tokenMap)))
```
Comment (laplaque) [inline anonymizer.go]: **[CRITICAL]** `TokensDeanonymized` is incremented at reader *creation* time — before the goroutine in `readLoop` has replaced a single byte. If the caller never reads the returned `ReadCloser` (e.g. early return on upstream error), tokens are counted but never de-anonymized. Move the increment into `processTextDelta` / `flushRemainder` at the point of actual substitution, matching `DeanonymizeText`.

File: config.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+	// CustomPIIPatterns are user-defined patterns scoped to a named pack.
+	CustomPIIPatterns []CustomPIIPattern `json:"customPIIPatterns"`
+}
+
+// protectedPIITypes are built-in PII type names that custom patterns must not use.
+var protectedPIITypes = map[string]bool{
+	"EMAIL": true, "PHONE": true, "SSN": true, "CC": true,
+	"IP": true, "APIKEY": true, "ADDRESS": true, "PERSON": true,
+	"ORG": true, "LOCATION": true, "MISC": true,
```
Comment (laplaque) [inline config.go]: **[SIGNIFICANT]** `protectedPIITypes` is missing every type introduced by the SECRETS pack: `SSHKEY`, `JWT`, `BEARER`, `DBCONN`, `AWSKEY`, `GHTOKEN`. A user-supplied custom pattern can declare `type: "JWT"` and silently shadow the built-in entry. Add all pack-defined types here.
